lightkurve/correctors/pldcorrector.py

```diff
@@ -139,144 +237,284 @@ def correct(self, aperture_mask=None, cadence_mask=None, gp_timescale=30,
-            Returns a corrected lightcurve object. Depending on the input, the
-            returned object will be a `KeplerLightCurve`, `TessLightCurve`, or
-            general `LightCurve` object.
+        clc : `.LightCurve`
+            Noise-corrected `.LightCurve`.
         """
-        if use_gp:
-            # Verify optional dependency
-            try:
-                import celerite
-            except ImportError:
-                log.error("PLD uses the `celerite` Python package. "
-                          "See the installation instructions at "
-                          "https://docs.lightkurve.org/about/install.html. "
-                          "`use_gp` has been set to `False`.")
-                use_gp = False
-
-        # Parse the aperture mask to accept strings etc.
-        aperture = self.tpf._parse_aperture_mask(aperture_mask)
-
-        # generate flux light curve from desired pixels
-        lc = self.tpf.to_lightcurve(aperture_mask=aperture)
-        rawflux = lc.flux.value
-        rawflux_err = lc.flux_err.value
-
-        # create nan mask
-        nanmask = np.isfinite(self.time)
-        nanmask &= np.isfinite(rawflux)
-        nanmask &= np.isfinite(rawflux_err)
-        nanmask &= np.abs(rawflux_err) > 1e-12
-
-        # mask out nan values
-        rawflux = rawflux[nanmask]
-        rawflux_err = rawflux_err[nanmask]
-        self.flux = self.flux[nanmask]
-        self.flux_err = self.flux_err[nanmask]
-        self.time = self.time[nanmask]
-
-        # parse the PLD aperture mask
-        pld_pixel_mask = self.tpf._parse_aperture_mask(pld_aperture_mask)
-
-        # find pixel bounds of aperture on tpf
-        xmin, xmax = min(np.where(pld_pixel_mask)[0]),  max(np.where(pld_pixel_mask)[0])
-        ymin, ymax = min(np.where(pld_pixel_mask)[1]),  max(np.where(pld_pixel_mask)[1])
+        if background_mask is None:
+            # Default to pixels <1-sigma above the background
+            background_mask = ~self.tpf.create_threshold_mask(1, reference_pixel=None)
+        self.background_mask = background_mask
+        self.restore_trend = restore_trend
+
+        if dm is None:
+            dm = self.create_design_matrix(background_mask=background_mask,
+                                           pld_aperture_mask=pld_aperture_mask,
+                                           pld_order=pld_order,
+                                           n_pca_terms=n_pca_terms,
+                                           pixel_components=pixel_components,
+                                           spline_n_knots=spline_n_knots,
+                                           spline_degree=spline_degree,
+                                           sparse=sparse)
+
+
+        clc = super(PLDCorrector, self).correct(dm, **kwargs)
+        if restore_trend:
+            clc += (self.diagnostic_lightcurves['spline'] - np.median(self.diagnostic_lightcurves['spline'].flux))
+        self.dm = dm
```

Comment:
Likewise, I recommend removing this to avoid confusion, because `RegressionCorrector.correct()` already stores the design matrix as `self.design_matrix_collection` (and `self.dmc` is available as a shorthand alias):
```suggestion
```